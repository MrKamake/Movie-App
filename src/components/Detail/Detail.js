import React, { Component } from 'react';
import './Detail.scss';
import { MOVIE_IMAGE_URL } from '../../constants/constant';
import Rating from 'react-rating';
import Noimage from '../../image/Noimage.png';
import Adult from '../../image/adult-icon.png';
import Star from '../../image/rating_star.png';
import EmptyStar from '../../image/rating_empty_star.png';

class Detail extends Component {
  componentDidMount() {
    const { match, getMovieDetail } = this.props;
    const movieId = match.params.movie_id;

    getMovieDetail(movieId);
  }

  render() {
    const { movieDetail, movieActorList } = this.props;

    return (
      <div className="Detail">
        <div className="movie-element">
          {movieDetail.poster_path ? (
            <img
              className="poster"
              src={MOVIE_IMAGE_URL + movieDetail.poster_path}
              alt={movieDetail.title}
            />
          ) : (
            <img className="poster" src={Noimage} alt="Noimage" />
          )}
          <div className="movie-detail-description">
            <div className="movie-detail-title-wrapper">
              <div className="movie-detail-title">{movieDetail.title}</div>
              <div className="movie-detail-date">
                ({String(movieDetail.release_date).slice(0, 4)})
              </div>
              <div className="movie-detail-adult">
                {movieDetail.adult && <img src={Adult} alt="Adult" />}
              </div>
            </div>
            <Rating
              className="movie-detail-rating"
              readonly
              initialRating={movieDetail.vote_average / 2}
              emptySymbol={<img src={EmptyStar} className="icon" alt="icon" />}
              fullSymbol={<img src={Star} className="icon" alt="icon" />}
            />
            <div className="movie-genre-list">
              {movieDetail.genres &&
                movieDetail.genres.map((movieGenre, idx) => (
                  <div className="movie-genre" key={idx}>
                    {movieGenre.name}
                  </div>
                ))}
            </div>
            <div className="movie-detail-overview-title">
              {movieDetail.overview && 'Overview'}
            </div>
            <p className="movie-detail-overview">{movieDetail.overview}</p>
          </div>
          <div className="cast-title">
            {movieActorList[0] && 'Top Billed Cast'}
          </div>
          <div className="actor-list-wrapper">
            <div className="cast-list">
              {movieActorList.map(actor => (
                <div className="actor-data" key={actor.credit_id}>
                  {actor.profile_path ? (
                    <img
                      src={MOVIE_IMAGE_URL + actor.profile_path}
                      alt={actor.title}
                    />
                  ) : (
                    <img src={Noimage} alt="Noimage" />
                  )}
                  <div className="actor-name">{actor.name}</div>
                  <div className="actor-character">{actor.character}</div>
                </div>
              ))}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Detail;
