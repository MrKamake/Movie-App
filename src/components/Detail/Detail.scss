.Detail {
  text-align: center;
  color: white;
  .movie-element {
    display: inline-block;
    margin: 40px 20px;
    text-align: left;
    width: 80%;
    color: white;
    .poster {
      width: 300px;
      margin-left: 20px;
    }
    .movie-detail-description {
      display: inline-block;
      padding-left: 50px;
      .movie-detail-title-wrapper {
        position: relative;
        .movie-detail-title {
          display: inline-block;
          font-size: 2.5rem;
          font-weight: bold;
        }
        .movie-detail-date {
          display: inline-block;
          font-size: 1.5rem;
          color: darkgray;
        }
        .movie-detail-adult {
          position: absolute;
          top: 0;
          right: 0;
          margin-left: 5px;
          img {
            width: 70px;
          }
        }
      }
      .movie-detail-rating {
        display: inline-block;
        img {
          width: 20px;
        }
      }
      .movie-genre-list {
        padding: 0 1px;
        .movie-genre {
          display: inline-block;
          margin: 5px;
          padding: 3px 10px;
          border-radius: 4px;
          font-size: 1.2rem;
          background: gray;
        }
      }
      .movie-detail-overview-title {
        padding-top: 150px;
        font-size: 1.4rem;
        font-weight: bold;
      }
      .movie-detail-overview {
        width: 900px;
        height: 100px;
        font-size: 1.1rem;
        overflow: hidden;
      }
    }
    .cast-title {
      margin-left: 20px;
      padding-top: 30px;
      font-size: 1.7rem;
      font-weight: bold;
    }
    .actor-list-wrapper {
      display: flex;
      justify-content: space-around;
      .cast-list {
        .actor-data {
          display: flex;
          float: left;
          flex-direction: column;
          margin: 20px;
          width: 150px;
          height: 300px;
          img {
            width: 100%;
            height: 75%;
          }
          .actor-name {
            padding: 10px 0;
            width: 100%;
            font-size: 1.1rem;
            font-weight: bold;
            overflow: hidden;
          }
          .actor-character {
            font-size: 0.9rem;
            overflow: hidden;
          }
        }
      }
    }
  }
}
