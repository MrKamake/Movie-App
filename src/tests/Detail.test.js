import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Detail from '../components/Detail/Detail';

configure({ adapter: new Adapter() });

describe('<Detail />', () => {
  let wrapper, props;

  beforeEach(() => {
    props = {
      match: {
        params: {
          movie_id: 123
        }
      },
      getMovieDetail: jest.fn(),
      genreList: ['Horor', 'Comic', 'Action'],
      movieDetail: {
        poster_path: '/asdfoo3.jpg',
        title: 'Life of vaco',
        vote_average: 4,
        release_date: 2019 - 12 - 11,
        genre_ids: [1, 2, 5],
        overview: 'askdfja2qbnelvpaoisfkw'
      },
      movieActorList: [
        {
          character: 'Augusto',
          credit_id: '5aa327c20e0a26075700097b',
          name: 'Felipe Camargo'
        },
        {
          character: 'Tereza',
          credit_id: '5aa327b2925141278d000773',
          name: 'Carol Castro'
        }
      ]
    };

    wrapper = shallow(<Detail {...props} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain element', () => {
    expect(
      wrapper.contains(<div className="movie-detail-title">Life of vaco</div>)
    ).toBe(true);
    expect(
      wrapper.contains(<div className="actor-name">Felipe Camargo</div>)
    ).toBe(true);
    expect(
      wrapper.contains(<div className="actor-character">Tereza</div>)
    ).toBe(true);
  });
});
