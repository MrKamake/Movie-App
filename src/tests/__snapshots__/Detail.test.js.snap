// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Detail /> matches snapshot 1`] = `
<div
  className="Detail"
>
  <div
    className="movie-element"
  >
    <img
      alt="Life of vaco"
      className="poster"
      src="https://image.tmdb.org/t/p/w300_and_h450_bestv2/asdfoo3.jpg"
    />
    <div
      className="movie-detail-description"
    >
      <div
        className="movie-detail-title-wrapper"
      >
        <div
          className="movie-detail-title"
        >
          Life of vaco
        </div>
        <div
          className="movie-detail-date"
        >
          (
          1996
          )
        </div>
        <div
          className="movie-detail-adult"
        />
      </div>
      <RatingAPILayer
        className="movie-detail-rating"
        direction="ltr"
        emptySymbol={
          <img
            alt="icon"
            className="icon"
            src="rating_empty_star.png"
          />
        }
        fractions={1}
        fullSymbol={
          <img
            alt="icon"
            className="icon"
            src="rating_star.png"
          />
        }
        initialRating={2}
        onChange={[Function]}
        onClick={[Function]}
        onHover={[Function]}
        placeholderSymbol={
          Object {
            "backgroundColor": "red",
            "border": "5px double white",
            "borderRadius": "50%",
            "display": "inline-block",
            "height": 30,
            "width": 30,
          }
        }
        quiet={false}
        readonly={true}
        start={0}
        step={1}
        stop={5}
      />
      <div
        className="movie-detail-overview-title"
      >
        Overview
      </div>
      <p
        className="movie-detail-overview"
      >
        askdfja2qbnelvpaoisfkw
      </p>
    </div>
    <div
      className="cast-title"
    >
       Top Billed Cast
    </div>
    <div
      className="actor-list-wrapper"
    >
      <div
        className="cast-list"
      >
        <div
          className="actor-data"
          key="5aa327c20e0a26075700097b"
        >
          <img
            alt="Noimage"
            src="Noimage.png"
          />
          <div
            className="actor-name"
          >
            Felipe Camargo
          </div>
          <div
            className="actor-character"
          >
            Augusto
          </div>
        </div>
        <div
          className="actor-data"
          key="5aa327b2925141278d000773"
        >
          <img
            alt="Noimage"
            src="Noimage.png"
          />
          <div
            className="actor-name"
          >
            Carol Castro
          </div>
          <div
            className="actor-character"
          >
            Tereza
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
